tests/MongoDB.Driver.Tests/MongoCollectionImplTests.cs

```diff
@@ -170,7 +174,11 @@ public void Settings_should_be_set()
             };
             var cancellationToken1 = new CancellationTokenSource().Token;
             var cancellationToken2 = new CancellationTokenSource().Token;
-            var renderedPipeline = RenderPipeline(subject, pipeline);
+            var expectedPipeline = new List<BsonDocument>(RenderPipeline(subject, pipeline).Documents);
+            if (!usingDifferentOutputDatabase)
+            {
+                expectedPipeline[1] = new BsonDocument("$out", outputCollection.CollectionNamespace.CollectionName);
+            }
```

Comment:
We have to be careful here because we do *not* want to modify existing pipeline stages in place because that alters the input to the test.

That's why `expectedPipeline` is a *new* `List` and if we need to simplify `$out` we create the entire new stage from scratch rather than modifying the existing one.